File: practical-05-caesar-cipher/caesar_cipher.py
def caesar_encrypt(plaintext, shift):
    """Encrypt plaintext using Caesar cipher with given shift."""
    result = ""
    for char in plaintext:
        if char.isupper():
            result += chr((ord(char) - 65 + shift) % 26 + 65)
        elif char.islower():
            result += chr((ord(char) - 97 + shift) % 26 + 97)
        else:
            result += char
    return result


def caesar_decrypt(ciphertext, shift):
    """Decrypt ciphertext using Caesar cipher with given shift."""
    return caesar_encrypt(ciphertext, 26 - shift)


if __name__ == "__main__":
    text = "Hello World"
    key = 3

    encrypted = caesar_encrypt(text, key)
    decrypted = caesar_decrypt(encrypted, key)

    print(f"Original: {text}")
    print(f"Encrypted (shift={key}): {encrypted}")
    print(f"Decrypted: {decrypted}")

File: c programs/caesar_cipher.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void caesar_encrypt(const char *plaintext, int shift, char *result) {
    int i = 0;
    for (i = 0; plaintext[i] != '\0'; i++) {
        if (isupper(plaintext[i])) {
            result[i] = (char)((plaintext[i] - 65 + shift) % 26 + 65);
        } else if (islower(plaintext[i])) {
            result[i] = (char)((plaintext[i] - 97 + shift) % 26 + 97);
        } else {
            result[i] = plaintext[i];
        }
    }
    result[i] = '\0';
}

void caesar_decrypt(const char *ciphertext, int shift, char *result) {
    caesar_encrypt(ciphertext, 26 - shift, result);
}

int main() {
    const char *text = "Hello World";
    int key = 3;
    
    char encrypted[100];
    char decrypted[100];
    
    caesar_encrypt(text, key, encrypted);
    caesar_decrypt(encrypted, key, decrypted);
    
    printf("Original: %s\n", text);
    printf("Encrypted (shift=%d): %s\n", key, encrypted);
    printf("Decrypted: %s\n", decrypted);
    
    return 0;
}
